### PR TITLE
Issue #2585419: Replace evaluate() with doEvaluate() for PathAliasExi…

### PR DESCRIPTION
**Pull request by emclaughlin1215**
…sts condition.

See d.o issues: https://www.drupal.org/node/2585419 and https://www.drupal.org/node/2572119

### CODE REVIEW

File: src/Plugin/Condition/PathAliasExists.php [hunk 1]
```diff
@@ -73,11 +74,19 @@ public static function create(ContainerInterface $container, array $configuratio
   }
 
   /**
-   * {@inheritdoc}
+   * Check if a path alias exists.
+   *
+   * @param string $alias
+   *   The alias to see if exists.
+   * @param mixed $language
```
Comment (klausi) [inline src/Plugin/Condition/PathAliasExists.php]: type hint should be \Drupal\Core\Language\LanguageInterface|null

File: src/Plugin/Condition/PathAliasExists.php [hunk 2]
```diff
@@ -73,11 +74,19 @@ public static function create(ContainerInterface $container, array $configuratio
...
+   *
+   * @return bool
+   *   TRUE if the system path does not match the given alias (ie: the alias
+   *   exists).
    */
-  public function evaluate() {
-    $alias = $this->getContextValue('alias');
-    $language = $this->getContext('language')->hasContextValue() ? $this->getContextValue('language')->getId() : NULL;
+  public function doEvaluate($alias, LanguageInterface $language = NULL) {
+    $language = !is_null($language) ? $language->getId() : NULL;
```
Comment (klausi) [inline src/Plugin/Condition/PathAliasExists.php]: no need for this line we can just pass on $language, the alias manager also accepts NULL.

Comment (emclaughlin1215) [inline src/Plugin/Condition/PathAliasExists.php]: Just passing on $language creates the same problem mentioned in PR 263. Specifically, getPathByAlias() wants a string value for the language ID, not the language object itself, or NULL.

Comment (klausi) [inline src/Plugin/Condition/PathAliasExists.php]: see comments in #263

**Comment by emclaughlin1215** [comment]
Updated based on comments.

File: src/Plugin/Condition/PathAliasExists.php [hunk 3]
```diff
@@ -73,11 +74,19 @@ public static function create(ContainerInterface $container, array $configuratio
...
+   *
+   * @return bool
+   *   TRUE if the system path does not match the given alias (ie: the alias
+   *   exists).
    */
-  public function evaluate() {
-    $alias = $this->getContextValue('alias');
-    $language = $this->getContext('language')->hasContextValue() ? $this->getContextValue('language')->getId() : NULL;
+  protected function doEvaluate($alias, LanguageInterface $language = NULL) {
+    $language = is_null($language) ? NULL : $language->getId();
```
Comment (klausi) [inline src/Plugin/Condition/PathAliasExists.php:89]: Let's not overwrite the parameter variable $language here but rather use $lang_code instead.

**Comment by klausi** [comment]
continued in #298

**Pull request closed without merging**